controllers/budget_controller.py

```diff
@@ -1,76 +1,145 @@
-        print("\n{}".format(budgets_list[0].name.upper()))  # Print first budget's name
 
-        # LOAD WHOLE BUDGET
+def change_budget(user_id):
+    print("\nYour budgets: ")
 
-        # loop through parent categories of the first budget in the list
-        for parent in budgets_list[0].parent_categories:
+    ''' Printing the table with user's budgets.'''
+    # Definition of PrettyTable for budgets.
+    table_show_budgets = PrettyTable()
+    table_show_budgets.field_names = ["id", "name"]
 
-            # Start calculating the parent's available amount based its children' available_amount
-            parent_available_sum = 0.00
+    # Reading data from databe and inserting into PrettyTable
+    for instance in session.query(Budget).filter(Budget.user_id == user_id):
+        table_show_budgets.add_row([Budget.give_budgets(instance)[0], Budget.give_budgets(instance)[1]])
 
-            # Loop through the parent's categories to add available_amount to the parent_available_sum
-            for category in parent.categories:
-                parent_available_sum += category.available_amount
-
-            # Format the result and print it along parent category's name
-            formatted_sum = "{:.2f} zł".format(parent_available_sum)
-            print("\n---------------- {}, dostępna kwota: {} ---------------- \n".format(parent.name, formatted_sum))
-
-            # Loop through the categories of the parent ONCE AGAIN, this time to print them
-            n = 1  # Position (number) of the category within the parent
-            for category in parent.categories:
-                formatted_available = "{:.2f} zł".format(category.available_amount)
-                print("{}. {}, dostępne środki: {}".format(n, category.name, formatted_available))
-                n += 1  # Increment the category number
-
-        print("\n")  # Print space between the next command
+    print(table_show_budgets)
 
+    ''' Printing the budget's table that user selected.'''
+    choice = input("Which budget to show? Input budget's id OR input 'n' to create new budget: ")
+    if choice == 'n':
+        budget_id = add_budget(user_id)
+    else:
+        # TODO: Add validation if the choice is correct
+        budget_id = choice
+    return budget_id
+
+
+def print_budget(budget_id):
+    # Definition of PrettyTable
+    table_budget = PrettyTable()
+    table_budget.field_names = [" id, CATEGORY", "BUDGETED", "ACTIVITY", "AVAILABLE"]
+    table_budget.align = "r"  # align in all columns to the right side
+    table_budget.align[" id, CATEGORY"] = "l"  # align in column "CATEGORY" to the left side
+    table_budget.float_format = "1.2"  # the way floating point data is printed
+
+    # Reading data from database and inserting into PrettyTable
+    sum_budgeted = 0
+    sum_available = 0
+    total_budgeted = 0
+    total_activity = 0
+    total_available = 0
+
+    budget = session.query(Budget).filter_by(id=budget_id).first()
```

Comment:
Tutaj pozwoliłem sobie uporządkować kwerki.

Pamiętajmy, że między funkcjami można sobie również podawać obiekty, a nie tylko id_obiektów

Reply:
Dodałem przekazywanie obiektów między funkcjami. Kilka kwerek też poprawiłem, ale nie wszystkie

Reply:
Ekstra, dzięki. To ja dziś poprawie pozostałe kwerki.